client/lib/reader-lists-items/store.js

```diff
@@ -0,0 +1,91 @@
+// Reader Lists Items Store
+
+// External dependencies
+import { List, fromJS } from 'immutable';
+import debugModule from 'debug';
+
+// Internal dependencies
+import { action as actionTypes } from './constants';
+import { createReducerStore } from 'lib/store';
+
+const debug = debugModule( 'calypso:reader-lists-items' ); //eslint-disable-line no-unused-vars
+
+const initialState = {
+	items: {},
+	errors: [],
+	currentPage: {},
+	isLastPage: {},
+	isFetching: false
+};
+const defaultListItem = List();
+
+function getListItems( state, listId ) {
+	return state.get( 'items' ).get( parseInt( listId ), defaultListItem ); // eslint-disable-line new-cap
+}
+
+function receiveItems( state, data ) {
+	// Is it the last page?
+	let isLastPage = state.get( 'isLastPage' );
+	if ( data.number === 0 ) {
+		isLastPage = isLastPage.set( data.list_ID, true );
+	}
+
+	// Add new items from response
+	let items = state.get( 'items' );
+	if ( data && data.items ) {
+		const existingItems = getListItems( state, data.list_ID );
+		items = items.setIn( [ data.list_ID ], existingItems.concat( fromJS( data.items ) ) );
+	}
+
+	// Set the current page
+	let currentPage = state.get( 'currentPage' );
+	currentPage = currentPage.set( data.list_ID, data.page );
+
+	return state.set( 'isLastPage', isLastPage ).set( 'items', items ).set( 'currentPage', currentPage );
```

Comment:
should probably use `withMutations( callback )` here to speed up the bulk change. See https://facebook.github.io/immutable-js/docs/#/Map/withMutations